resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsLifecycleManager.scala

```diff
@@ -176,6 +177,7 @@ private[spark] class ExecutorPodsLifecycleManager(
 
   private def findExitCode(podState: FinalPodState): Int = {
     podState.pod.getStatus.getContainerStatuses.asScala.find { containerStatus =>
+      containerStatus.getName == DEFAULT_EXECUTOR_CONTAINER_NAME &
```

Comment:
Why this change? It seems like if someone manually registered executors with us this might break. That's not as hypothetical as it sounds, since some folks (like I believe some of the Spark CRDs) launch their own executors.

Reply:
CC @mikaylakonst

Reply:
This make sure the exit code shows in the pod status is reflect from the executor container.